src/hera/workflows/script.py

```diff
@@ -51,92 +69,36 @@ class Script(
+        raise ValueError(f"Unknown constructor {v}")
+
+    @validator("command", always=True)
+    @classmethod
+    def _set_command(cls, v):
+        return v or global_config.script_command
 
-        Returns
-        -------
-        str
-            Final formatted script.
-        """
-        if callable(self.source):
-            signature = inspect.signature(self.source)
-            args = inspect.getfullargspec(self.source).args
-            if signature.return_annotation == str:
-                # Resolve function by filling in templated inputs
-                input_params_names = [p.name for p in self.inputs if isinstance(p, Parameter)]  # type: ignore
-                missing_args = set(args) - set(input_params_names)
-                if missing_args:
-                    raise ValueError(f"Missing inputs for source args: {missing_args}")
-                kwargs = {name: f"{{{{inputs.parameters.{name}}}}}" for name in args}
-                # Resolve the function to a string
-                return self.source(**kwargs)
-            else:
-                script = ""
-                # Argo will save the script as a file and run it with cmd:
-                # - python /argo/staging/script
-                # However, this prevents the script from importing modules in its cwd,
-                # since it's looking for files relative to the script path.
-                # We fix this by appending the cwd path to sys:
-                if self.add_cwd_to_sys_path:
-                    script = "import os\nimport sys\nsys.path.append(os.getcwd())\n"
-
-                script_extra = self._get_param_script_portion() if args else None
-                if script_extra:
-                    script += copy.deepcopy(script_extra)
-                    script += "\n"
-
-                # content represents the function components, separated by new lines
-                # therefore, the actual code block occurs after the end parenthesis, which is a literal `):\n`
-                content = inspect.getsourcelines(self.source)[0]
-                token_index, start_token = 1, ":\n"
-                for curr_index, curr_token in enumerate(content):
-                    if start_token in curr_token:
-                        # when we find the curr token we find the end of the function header. The next index is the
-                        # starting point of the function body
-                        token_index = curr_index + 1
-                        break
-
-                s = "".join(content[token_index:])
-                script += textwrap.dedent(s)
-                return textwrap.dedent(script)
-        else:
-            assert isinstance(self.source, str)
-            return self.source
+    @root_validator
+    @classmethod
+    def _constructor_validate(cls, values):
+        constructor = values.get("constructor")
+        assert isinstance(constructor, ScriptConstructor)
+        return constructor.transform_inputs(cls, values)
```

Comment:
This allows users to modify input args based on the constructor type. This is the first script constructor hook.